diagnosticsModule/Public/Join-DiagnosticsFiles.ps1

```diff
@@ -0,0 +1,106 @@
+            
+            $fileName = $_ | Select-Object -ExpandProperty FullName
+
+            Write-Host "  - Merging file: $fileName"  
+            
+            $allTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select AllTests
+            $passedTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select PassedTests
+            $warningTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select WarningTests
+            $failedTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select FailedTests
+            $errorTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select ErrorTests
+            $notRunTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select NotRunTests
+            $reachableServers = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select ReachableServers
+            $unreachableServers = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select UnreachableServers
+
+            $mergedTestData["AllTests"] += $allTests.AllTests
+            $mergedTestData["PassedTests"] += $passedTests.PassedTests
+            $mergedTestData["WarningTests"] += $warningTests.WarningTests
+            $mergedTestData["FailedTests"] += $failedTests.FailedTests
+            $mergedTestData["ErrorTests"] += $errorTests.ErrorTests
+            $mergedTestData["NotRunTests"] += $notRunTests.NotRunTests
+            $mergedTestData["ReachableServers"] += $reachableServers.ReachableServers
+            $mergedTestData["UnreachableServers"] += $unreachableServers.UnreachableServers
+
+            $configuration = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Adfs-Configuration
+
+            # Only add the ADFS Configuration information if this is coming from the primary node
+            if (($configuration | Select-Object -ExpandProperty Role).ToString() -eq "1") {
+                $adfsConfiguration = $configuration
+            }
+
+            $version = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object Version | Select Version
+        }
+        
+    $outputPath = $FilePath + $OutputFileName
+    Write-Host
```

Comment:
extra?

Reply:
adds an extra blank line.  but looks fine without it, will remove.